src/keylessBackup/PhoneBackupInput.tsx

```diff
@@ -0,0 +1,164 @@
+      cachedCountryCallingCode || '',
+      selectedCountryCodeAlpha2 || RNLocalize.getCountry()
+    )
+  )
+  const country = phoneNumberInfo.countryCodeAlpha2
+    ? countries.getCountryByCodeAlpha2(phoneNumberInfo.countryCodeAlpha2)
+    : undefined
+
+  useEffect(() => {
+    const newCountryAlpha2 = route.params?.selectedCountryCodeAlpha2
+    if (newCountryAlpha2 && newCountryAlpha2 !== phoneNumberInfo.countryCodeAlpha2) {
+      const countryCallingCode =
+        countries.getCountryByCodeAlpha2(newCountryAlpha2)?.countryCallingCode ?? ''
+      setPhoneNumberInfo(
+        getPhoneNumberState(
+          phoneNumberInfo.internationalPhoneNumber,
+          countryCallingCode,
+          newCountryAlpha2
+        )
+      )
+    }
+  }, [route.params?.selectedCountryCodeAlpha2])
+
+  const onChangePhoneNumberInput = (
+    internationalPhoneNumber: string,
+    countryCallingCode: string
+  ) => {
+    setPhoneNumberInfo(
+      getPhoneNumberState(
+        internationalPhoneNumber,
+        countryCallingCode,
+        phoneNumberInfo.countryCodeAlpha2
+      )
+    )
+  }
+  const onPressCountry = () => {
+    navigate(Screens.SelectCountry, {
+      countries,
+      selectedCountryCodeAlpha2: phoneNumberInfo.countryCodeAlpha2,
+      hideOnboardingStep: true,
+      keylessBackupFlow,
+    })
+  }
+
+  const onPressContinue = () => {
+    ValoraAnalytics.track(KeylessBackupEvents.enter_phone_number_continue, {
+      keylessBackupFlow,
+    })
+    // TODO: handle the rest of the phone backup flow
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView style={styles.scrollContainer}>
+        <Text style={styles.title}>
+          {keylessBackupFlow === KeylessBackupFlow.Setup
+            ? t('signInWithPhone.title')
+            : t('signInWithPhone.titleRestore')}
+        </Text>
+        <Text style={styles.subtitle}>
+          {keylessBackupFlow === KeylessBackupFlow.Setup
+            ? t('signInWithPhone.subtitle')
+            : t('signInWithPhone.subtitleRestore')}
+        </Text>
+        <PhoneNumberInput
+          countryFlagStyle={styles.countryFlagStyle}
+          label={''}
+          country={country}
+          internationalPhoneNumber={phoneNumberInfo.internationalPhoneNumber}
+          onPressCountry={onPressCountry}
+          onChange={onChangePhoneNumberInput}
+        />
+      </ScrollView>
+      <Button
+        testID="PhoneBackupInput/Continue"
+        onPress={onPressContinue}
+        text={keylessBackupFlow === KeylessBackupFlow.Setup ? t('continue') : t('next')}
+        size={BtnSizes.FULL}
+        type={BtnTypes.ONBOARDING}
+        style={styles.button}
+        disabled={!phoneNumberInfo.isValidNumber}
+      />
+    </SafeAreaView>
+  )
+}
+
+PhoneBackupInput.navigationOptions = () => ({
+  ...emptyHeader,
+  headerLeft: () => (
+    <TopBarIconButton
+      style={styles.cancelButton}
+      icon={<Times height={16} />}
+      onPress={navigateBack}
```

Comment:
This would go back to the email sign in screen right? I think right now the X on the email screen is set to navigate to settings.. Although I think that won't work for the restore flow.

Reply:
I added a small addendum to this ticket to handle the X button on all the keylessbackup screens. https://linear.app/valora/issue/ACT-770/warning-modal-for-skipexit-backup-flow
I'll add a todo here